scripts/BuildAndTest.ps1

```diff
@@ -127,12 +150,25 @@ function Set-SarifFileAssociationRegistrySettings {
     }
 }
 
-& $PSScriptRoot\BeforeBuild.ps1 -NoClean:$NoClean -NoRestore:$NoRestore
-if (-not $?) {
-    Exit-WithFailureMessage $ScriptName "BeforeBuild failed."
+if (-not $NoClean) {
```

Comment:
>i [](start = 0, length = 1)

I inlined the contents of `BeforeBuild.ps1`. That file was a relic of the time when AppVeyor didn't allow you to specify a full build script, only the name of a project to build + a "pre-build" script. Now AppVeyor lets you specify a full build script. Anyway, we'll move away from AppVeyor in a future PR.